pkg/apis/hive/v1alpha1/hiveconfig_types.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// HiveConfigSpec defines the desired state of Hive
+type HiveConfigSpec struct {
```

Comment:
I had a todo to hook it up such that deleting the HiveConfig would clean up the install, but I couldn't see how it would be useful to anyone just yet. A state field might be a better approach. Make sense for future enhancement or would it be beneficial to anyone right now?

Reply:
future enhancement is fine